saq/queue/postgres.py

```diff
@@ -0,0 +1,447 @@
+                    SELECT count(*) FROM {JOBS_TABLE}
+                    WHERE status IN ('new', 'deferred', 'queued', 'active')
+                      AND queue = %(queue)s
+                    """,
+                    {"queue": self.name},
+                )
+            else:
+                raise ValueError("Can't count unknown type {kind}")
+
+            result = await cursor.fetchone()
+            assert result
+            return result[0]
+
+    async def schedule(self, lock: int = 1) -> t.Any: ...
+
+    async def sweep(self, lock: int = 60, abort: float = 5.0) -> list[t.Any]:
+        """Not applicable to Postgres queue"""
+        return []
+
+    async def listen(
+        self,
+        job_keys: Iterable[str],
+        callback: ListenCallback,
+        timeout: float | None = 10,
+    ) -> None:
+        async with self.pool.connection() as conn:
+            for key in job_keys:
+                await conn.execute(f'LISTEN "{key}"')
+            await conn.commit()
+            gen = conn.notifies(timeout=timeout or None)
+            async for notify in gen:
+                payload = self._load(notify.payload)
+                key = payload["key"]
+                status = Status[payload["status"].upper()]
+                if asyncio.iscoroutinefunction(callback):
+                    stop = await callback(key, status)
+                else:
+                    stop = callback(key, status)
+
+                if stop:
+                    await gen.aclose()
+
+    async def notify(self, job: Job) -> None:
+        async with self._get_connection(job.key) as conn:
+            payload = self._dump({"key": job.key, "status": job.status})
+            await conn.execute(f"NOTIFY \"{job.key}\", '{payload}'")
+
+    async def update(self, job: Job) -> None:
+        job.touched = now()
+        async with self._get_connection(job.key) as conn:
+            await conn.execute(
+                f"""
+                    UPDATE {self.jobs_table} SET job = %(job)s, status = %(status)s
+                    WHERE key = %(key)s
+                    """,
+                {"job": self.serialize(job), "status": job.status, "key": job.key},
+            )
+        await self.notify(job)
+
+    async def job(self, job_key: str) -> Job | None:
+        async with self._get_connection(job_key) as conn, conn.cursor() as cursor:
+            await cursor.execute(
+                f"""
+                SELECT job FROM {self.jobs_table}
+                WHERE key = %(key)s
+                """,
+                {"key": job_key},
+            )
+            job = await cursor.fetchone()
+            if job:
+                return self.deserialize(job[0])
+        return None
+
+    async def abort(self, job: Job, error: str, ttl: float = 5) -> None:
+        job.error = error
+        job.status = Status.ABORTING
+
+        await self.update(job)
+        asyncio.create_task(self.delete(job, status=Status.ABORTED, delay=ttl))
```

Comment:
why create_task here?

where is the code that actually kills the task?

Reply:
This deletes aborted jobs from the jobs table. The worker contains the code that kills the task.

Reply:
I think we need a janitor task to clear out jobs from the jobs table based on ttl, so this can go away.